api/bootstrap/kubeadm/v1beta2/kubeadmconfigtemplate_types.go

```diff
@@ -56,7 +58,7 @@ type KubeadmConfigTemplate struct {
 
 	// spec is the desired state of KubeadmConfigTemplate.
 	// +optional
-	Spec KubeadmConfigTemplateSpec `json:"spec,omitempty"`
+	Spec KubeadmConfigTemplateSpec `json:"spec,omitempty,omitzero"`
```

Comment:
I think here the Spec is actually optional. See for example: https://storage.googleapis.com/kubernetes-ci-logs/logs/periodic-cluster-api-e2e-upgrade-1-28-1-29-main/1939753255810633728/artifacts/clusters/bootstrap/resources/k8s-upgrade-and-conformance-nnfo3e/KubeadmConfigTemplate/k8s-upgrade-and-conformance-vchn5p-md-0-pqlxv.yaml

Nothing there in the spec is required

Reply:
No they are not defaulted, but we set them because we need them in the test. But I think in reality there is none of them that actually has to be there with a real infra provider

> What is the purpose of creating this object?

You have to reference something in the MachineDeployment and I can't change that easily